core/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/list/ListTasksRequest.java

```diff
@@ -47,15 +48,31 @@ public ListTasksRequest detailed(boolean detailed) {
         return this;
     }
 
+    /**
+     * Should this request wait for all found tasks to complete?
+     */
+    public boolean shouldWaitForCompletion() {
+        return waitForCompletion;
+    }
+
+    /**
+     * Should this request wait for all found tasks to complete?
+     */
+    public void setWaitForCompletion(boolean waitForCompletion) {
```

Comment:
Somehow we need to distinguish "background" tasks for the cluster from those started by rest actions.